File: Leetcode/448/448.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        constexpr int EXIST = 0x3f3f3f3f;
        int n = (int)nums.size();
        int key;
        for (int i = 0; i < n; ++i) {
            key = nums[i];
            // restore
            if (key > EXIST) {
                key -= EXIST;
            }
            // the key already exists
            if (nums[key-1] > EXIST) {
                continue;
            }
            nums[key-1] += EXIST;
        }
        vector<int> ans;
        for (int i = 0; i < n; ++i) {
            if (nums[i] <= EXIST) {
                ans.emplace_back(i+1);
            }
        }
        return ans;
    }
};

File: Leetcode/448/448.go
const EXIST = 0x3f3f3f3f

func restore(n int) int {
	if n >= EXIST {
		n -= EXIST
	}
	return n
}

func findDisappearedNumbers(nums []int) []int {
	n := len(nums)
	nums = append(nums, 0)
	for i := 0; i < n; i++ {
        if nums[restore(nums[i])] < EXIST {
			nums[restore(nums[i])] += EXIST
		}
	}
	var ans []int
	for i := 1; i <= n; i++ {
		if nums[i] < EXIST {
			ans = append(ans, i)
		}
	}
	return ans
}